### PR TITLE
Expose/test span-based Version methods

### PR DESCRIPTION
**Pull request by stephentoub**
Depends on https://github.com/dotnet/coreclr/pull/13424
Fixes dotnet/corefx#22376
cc: @AlexGhiondea, @joperezr

### CODE REVIEW

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by stephentoub** [comment]
@dotnet-bot test this please

File: src/System.Runtime/ref/System.Runtime.cs [hunk 1]
```diff
@@ -3556,9 +3556,13 @@ public sealed partial class Version : System.IComparable, System.IComparable<Sys
         public static bool operator <(System.Version v1, System.Version v2) { throw null; }
         public static bool operator <=(System.Version v1, System.Version v2) { throw null; }
         public static System.Version Parse(string input) { throw null; }
+        public static System.Version Parse(ReadOnlySpan<char> input) { throw null; }
         public override string ToString() { throw null; }
         public string ToString(int fieldCount) { throw null; }
+        public bool TryFormat(System.Span<char> destination, out int charsWritten) { throw null; }
```
Comment (joperezr) [inline src/System.Runtime/ref/System.Runtime.cs]: More of a general question for @weshaggard: This ref currently builds for netcoreapp and uap, do we want to isolate all of these Span methods to be netcoreapp only? Or do we expect to add Span support to uap in the future?

Comment (weshaggard) [inline src/System.Runtime/ref/System.Runtime.cs]: Ideally we wouldn't fork between netcoreapp and uap, but I don't know the current state of Span support in uap. @stephentoub probably knows more then I do on that front.

Comment (stephentoub) [inline src/System.Runtime/ref/System.Runtime.cs]: Span is available in all of these, and we're ensuring that all of the new span-based APIs we're adding are added to both coreclr and corert.

**Comment by stephentoub** [comment]
@dotnet-bot test UWP NETNative x86 Release Build please
@dotnet-bot test Packaging All Configurations x64 Debug Build please

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Pull request merged**